nodes/filter/single.go

```diff
@@ -1,27 +1,345 @@
+	outputs := node.BuildOutputs(node.BuildOutput(node.Out, node.TypeFloat, nil, body.Outputs))
+	body = node.BuildNode(body, inputs, outputs, nil)
+	body.SetHelp(fmt.Sprintln("This node filters ‘input’ values. Only Numeric ‘input’ values greater than ‘OutMax’ are passed to ‘output’. Output the last known input float value if the input is less than 'OutMax'. Out put nil if any of the inputs are nil."))
+	return &OnlyGreater{body, 0}, nil
+}
+
+func NewOnlyLower(body *node.Spec) (node.Node, error) {
+	body = node.Defaults(body, onlyLower, category)
+	inputs := node.BuildInputs(node.BuildInput(node.In, node.TypeFloat, nil, body.Inputs), node.BuildInput(node.InMax, node.TypeFloat, nil, body.Inputs))
+	outputs := node.BuildOutputs(node.BuildOutput(node.Out, node.TypeFloat, nil, body.Outputs))
+	body = node.BuildNode(body, inputs, outputs, nil)
+	body.SetHelp(fmt.Sprintln("This node filters ‘input’ values. Only Numeric ‘input’ values less than 'InMax' are passed to ‘output’. Output the last known input float value if the input is greater than 'OutMin'. Out put nil if any of the inputs are nil."))
+	return &OnlyLower{body, 0}, nil
 }
 
-func (inst *OnlyTrue) Process() {
-	v, _ := inst.ReadPinAsBool(node.In)
-	if v {
-		inst.WritePinTrue(node.Out)
+func NewPreventDuplicates(body *node.Spec) (node.Node, error) {
+	body = node.Defaults(body, preventDuplicates, category)
+	inputs := node.BuildInputs(node.BuildInput(node.In, node.TypeFloat, nil, body.Inputs))
+	outputs := node.BuildOutputs(node.BuildOutput(node.Out, node.TypeFloat, nil, body.Outputs))
+	body = node.BuildNode(body, inputs, outputs, nil)
+	body.SetHelp(fmt.Sprintln("This node filters ‘input’ values. All ‘input’ values are passed to ‘output’ EXCEPT ‘input’ values which are equal to the previous ‘input’ value. Output the last known input float value if the subsequent input value is the same. Out put nil if the input is nil."))
+	return &PreventDuplicates{body, 0}, nil
+}
+
+// func (inst *OnlyTrue) Process() {
+// 	v, _ := inst.ReadPinAsBool(node.In)
+// 	if v {
+// 		inst.WritePinTrue(node.Out)
+// 	} else {
+// 		inst.WritePinFalse(node.Out)
+// 	}
+// }
+
+// func (inst *OnlyFalse) Process() {
+// 	v, null := inst.ReadPinAsBool(node.In)
+// 	if !v || null {
+// 		inst.WritePinFalse(node.Out)
+// 	} else {
+// 		inst.WritePinTrue(node.Out)
+// 	}
+// }
+
+// preventNull, output the last known value if the input became null.
+// last value is defaulted to be false
+func (inst *PreventNull) Process() {
+	v, null := inst.ReadPinAsBool(node.In)
+
+	if null {
+		inst.WritePinBool(node.Out, inst.lastValue)
 	} else {
-		inst.WritePinFalse(node.Out)
+		inst.WritePinBool(node.Out, v)
+		inst.lastValue = v
 	}
+}
+
+// preventEqualFloat, output the last known value if two inputs equal to each other. Out put nil if any of the inputs are nil.
+// lastValue is defaulted to be 0
+func (inst *PreventEqualFloat) Process() {
```

Comment:
Please name your variables so that they are easier for the next person to follow.  I would use `input` and `match` or something like that.  `v` and `m` are nice and short, but make it a pain for someone else to follow.